### ISSUE TITLE
Mbox.hs Option to Decrement Count by 1

### ISSUE THREAD

**Issue by dovej** [jaor/xmobar#326]
I have the Mbox plugin pointing at the local mbox file on my machine. Unfortunately, this file always has a "FOLDER INTERNAL DATA" message in it by default, which means the Mbox plugin always sees at least 1 message. Ideally, it'd be nice to have some option that ignores this message as most mail clients do. Simply subtracting 1 from the count would do.

There is a -u option that gets some of the functionality I'm looking for. When setting this option, the mail count won't display when there's only 1 message, so my xmobar stays uncluttered when the mbox only has the "FOLDER INTERNAL DATA" message in it. I'm not sure what this flag was intended for, as it is not documented.

**Comment by dovej** [comment]
From MAILER-DAEMON Fri Jan 12 00:02:34 2018
Date: 12 Jan 2018 00:02:34 -0500
From: Mail System Internal Data <MAILER-DAEMON@timeron>
Subject: DON'T DELETE THIS MESSAGE -- FOLDER INTERNAL DATA
Message-ID: <1515733354@timeron>
X-IMAP: 1515730556 0000000004
Status: RO

This text is part of the internal format of your mail folder, and is not
a real message.  It is created automatically by the mail system software.
If deleted, important folder data will be lost, and it will be re-created
with the data reset to initial values.

**Comment by jaor** [comment]
We could add a new option to specify senders to be ignored in the counts, and one could use it as

         ["-x", "MAILER-DAEMON", ...]

would that work? sounds a bit better than an adhoc "subtract one"... although the latter is easier to implement, and this idea will break if MAILER-DAEMON likes to send other messages that are important to count... does it?

**Comment by dovej** [comment]
I agree that my suggestion in the issue title is extremely ad-hoc and there is probably a wiser way to handle it.

I've narrowed down the root cause to alpine, which stores certain meta-data in these messages. alpine has a flag to disable these messages but at the cost of certain functionality. In my case, it turned out to be in my interest to switch to mutt for other reasons, which doesn't have this behavior. So, the issue is resolved for me and I am not particularly opinionated how this ticket is handled going forward (I'm fine if you just want to close it), but a very reasonable possibility remains that this will effect other users that have less flexibility than me to avoid it. For instance, I think these messages are also endemic to certain IMAP configurations, and at the very least certain pine/alpine users will be forced to live with them.

Aside, I'm still a little curious what the undocumented -u flag was intended for if not this :P

**Comment by jaor** [comment]
Hmm, i think i'll leave this open for now then, and see if anyone else
chimes in with similar needs so that we can find a good enough solution
for the general case.

`-u` is meant to show only the mailbox labels (when they have messages),
without the mail counts (barring bugs, of course).

**Comment by dovej** [comment]
Sounds good. Thanks for going through this with me, and sorry for adding to the pool only to bail ;)

Ah, that makes sense. I think it currently will show the counts as long as they are greater than 1.
